src/components/networked-hand-controls.js

```diff
@@ -336,29 +336,34 @@ AFRAME.registerComponent('networked-hand-controls', {
     // handling for older controller components (like vive-controls) that lack the cleaner hooks 
     // to block local post tracking. We allow pose tracking to start (so we can get loaded model), 
     // and then stop it. Note: we do not attempt to handle older tracked-controls-webvr.
-    let counter = 0;
-    let disableInterval = setInterval(() => {
-      counter++;
-      if (this.el.components['tracked-controls-webxr']) {
-        // there is no remove on this function in current iteration, so we kill it with pause and tick first.
-        this.el.components['tracked-controls-webxr'].pause(); 
-        this.el.components['tracked-controls-webxr'].tick = () => {};
-        this.el.removeAttribute('tracked-controls-webxr'); // just some cleanup for inspector, etc.
+    let intervalCounter = 0;
 
-        clearInterval(disableInterval);
-        this.el.play();
-      }
-      if (counter > 1000) {
-        // 25 seconds, don't want the possiblity of this running forever in the background
-        clearInterval(disableInterval);
+    const disablePostTracking = () => {
+      intervalCounter++;
+      if (intervalCounter > 1000) { // 7-17 seconds, assuming 140hz-60hz refresh rate
+        // don't want the possiblity of this running forever in the background
         try {
+          this.el.components[this.data.controllerComponent].pause();
           this.el.components[this.data.controllerComponent].remove();
         } catch(e) {
           console.error("error removing controller component", e);
         }
         throw new Error("never found tracked-controls-webxr, cancelling interval that was waiting to remove it and removing controller components")
       }
-    },25)
+      else if (this.el.components['tracked-controls-webxr']) {
+        // there is no remove on this component in current version of A-Frame, so we kill it with pause and tick first.
+        this.el.components['tracked-controls-webxr'].pause();  // remove its event listeners
+        this.el.components['tracked-controls-webxr'].tick = () => {}; // remove on-tick pose tracking in the component
```

Comment:
I don't understand why patching tick is needed here. Did you encounter an issue that would have required this?
Removing the component calls pause and remove and the pause is a wrapped method that calls sceneEl.removeBehavior(component) that remove calling the tick method from the animation loop. From what I understand all that is done synchronously in the same tick.

Reply:
The interesting parts in aframe code
https://github.com/aframevr/aframe/blob/795edfd65841d54aa2b186a02f660ffee606b50c/src/core/component.js#L671-L672
https://github.com/aframevr/aframe/blob/795edfd65841d54aa2b186a02f660ffee606b50c/src/core/component.js#L763-L794
https://github.com/aframevr/aframe/blob/837023cca1a04b408b4acde4c09823e09dc58e6d/src/core/a-entity.js#L381-L405

Reply:
A behavior is a component that has a tick or tock method
https://github.com/aframevr/aframe/blob/795edfd65841d54aa2b186a02f660ffee606b50c/src/core/component.js#L753-L755
Here is the scene tick that execute each behavior tick if isPlaying
https://github.com/aframevr/aframe/blob/0dd84abe8fd25367d87da207ed9e4bbcd6f9d92d/src/core/scene/a-scene.js#L731-L740

Reply:
Just an artifact of the way I developed an approach to make this possible, along with me not knowing for sure what guarantees I had about this stuff. I've cleaned up this section of code, will commit soon.